src/apps/external_executor/external_executor_indexing.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "external_executor_indexing.h"
+
+namespace externalexecutor
+{
+  MapIndex::MapIndex(
+    const std::shared_ptr<ccf::indexing::AbstractLFSAccess>& lfs_access_,
+    const std::string& map) :
+    lfs_access(lfs_access_),
+    indexed_data(lfs_access_, 10, map),
+    map_name(map)
+  {}
+
+  std::optional<std::string> MapIndex::fetch_data(std::string& key)
+  {
+    bool complete = true;
+
+    while (true)
```

Comment:
What's this while loop looping over? I don't see any range operations here.

Reply:
Removed